index.d.ts

```diff
@@ -61,7 +63,7 @@ declare module 'fastify' {
 
   interface FastifyRequest {
     auth0Verify: Auth0Verify
-    jwtDecode: typeof decode
+    jwtDecode: (options?: DecodeOptions & { complete: true } | DecodeOptions & { json: true } | DecodeOptions) => null | JwtPayload | Jwt
```

Comment:
Could the returned object be a string too?

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f4b63e02370940350887eaa82ac976dc2ecbf313/types/jsonwebtoken/index.d.ts#L243

Reply:
According to the type definition you linked it could be. 

But take a look at the implementation in the library:
https://github.com/auth0/node-jsonwebtoken/blob/master/decode.js#L10

It seems that whenever the payload is a string it will try to JSON.parse it. 
So i think the answer is: **no**, it will not be a string.